### DISCUSSION TITLE
Compatibility with Framer Motion?

### DISCUSSION THREAD

**Discussion by Jurek-Raben** [TanStack/router#576] [Q&A]
Hi,

the animation library Framer Motion is nicely compatible with React Router, so you can define transition animation while a route change. This is described for example here: https://www.section.io/engineering-education/page-transition-in-react-using-framer-motion/

Is this also possible with Tanstack Router? Will the AnimatePresence component of Framer Motion be compatible with Tanstack Router?

Also I am a bit confused by the version numbering in this project: v0.0.1 sounds like it is completely early bird, and might be completely changed with v0.0.2. So do you recommend to already use this library?

Thanks for insight.

**Comment by egor-xyz** [comment]
If someone solved it, can you share a solution for how to use framer-motion AnimatePresence with the router? (Not in nextJS)
